### ISSUE TITLE
src/cpp/server/server.cc fails to compile under gcc 4.4

### ISSUE THREAD

**Issue by dgquintas** [grpc/grpc#2936]
```
g++-4.4 -std=c++0x -pthread -I/usr/local/include   -Ithird_party/openssl/include -g -Wall -Wextra -Werror -Wno-long-long -Wno-unused-parameter -O2 -fPIC -I. -Iinclude -I./gens -DNDEBUG -DINSTALL_PREFIX=\"/usr/local\" -DGRPC_HAVE_PERFTOOLS    -MMD -MF objs/opt/src/cpp/server/server.dep -c -o objs/opt/src/cpp/server/server.o src/cpp/server/server.cc
src/cpp/server/server.cc: In member function ‘bool grpc::Server::Start()’:
src/cpp/server/server.cc:289: error: no matching function for call to ‘std::list<grpc::Server::SyncRequest, std::allocator<grpc::Server::SyncRequest> >::emplace_back(grpc::RpcServiceMethod*, const grpc::<anonymous class>&)’
make: *** [objs/opt/src/cpp/server/server.o] Error 1
```

**Comment by dgquintas** [comment]
@vjpai FYI given your recent dealings in these matter

**Comment by vjpai** [comment]
Ok. More motivation for getting Jenkins in for gcc 4.4

**Comment by vjpai** [comment]
I'll take this back from Yang. I've got an instance open with this setup, so I can test it quickly.

**Comment by vjpai** [comment]
Got it. emplace_back template on gcc4.4 doesn't like interacting with our nullptr anonymous class.
I'll send in a PR shortly.

**Comment by vjpai** [comment]
Ironically, this would have been fine if it weren't for my comment on Yang's PR.

**Comment by ctiller** [comment]
Is it time to kill nullptr?

**Comment by vjpai** [comment]
"Delenda est Carthago" ? I'm just worried about throwing out the baby with the bathwater. Getting this into Jenkins would make sure that we don't merge without building it there, and that would have spotted this error. Let's evaluate.

**Comment by ctiller** [comment]
Yeah, just worried about increasingly complicated workarounds when we could
just not use nullptr.

**Comment by vjpai** [comment]
Fixed by #2939
